arch/arm/mach-imx/imx8m/soc.c

```diff
@@ -608,6 +608,29 @@ enum boot_device get_boot_device(void)
 }
 #endif
 
+int boot_mode_getprisec(void)
+{
+	struct src *psrc = (struct src *)SRC_BASE_ADDR;
+
+	return !!(readl(&psrc->gpr10) & SRC_GPR10_PERSIST_SECONDARY_BOOT);
+}
+
+void boot_mode_set_secondary(void)
```

Comment:
@ricardosalveti @igoropaniuk how would we feel about combining  this into:
```
void boot_mode_enable_secondary(unsigned char enable)
{
	u32 persist_sec = SRC_GPR10_PERSIST_SECONDARY_BOOT;
	struct src *psrc = (struct src *)SRC_BASE_ADDR;

	if (enable)
		setbits_le32(&psrc->gpr10, persist_sec);
	else
		clrbits_le32(&psrc->gpr10, persist_sec);
}
```

Reply:
I've added the function you suggested, with only change
s/unsigned char enable/bool/g